src/main/java/janggi/board/Board.java

```diff
@@ -0,0 +1,70 @@
+package janggi.board;
+
+import janggi.Team;
+import janggi.board.position.Position;
+import janggi.piece.Piece;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Board {
+    private final Map<Position, Piece> board;
+
+    public Board(Map<Position, Piece> board) {
+        this.board = new HashMap<>(board);
+    }
+
+    public void movePiece(Position start, Position goal, Team team) {
+        Piece piece = board.get(start);
+        if (piece == null) {
+            throw new IllegalArgumentException("[ERROR] 출발 지점에 기물이 존재하지 않습니다.");
+        }
+        if (piece.isDifferentTeam(team)) {
+            throw new IllegalArgumentException("[ERROR] 같은 진영의 기물만 움직일 수 있습니다.");
+        }
+        piece.validateMovable(this, start, goal);
+        Piece attacked = move(start, goal);
+        if (attacked != null && attacked.isGeneral()) {
+            throw new GameOverException();
+        }
+    }
+
+    private Piece move(Position start, Position goal) {
+        Piece piece = board.remove(start);
+        return board.put(goal, piece);
+    }
+
+    public Map<Position, Piece> getBoard() {
+        return Collections.unmodifiableMap(board);
+    }
+
+    public boolean isPieceExists(Position position) {
+        return board.get(position) != null;
```

Comment:
```suggestion
        return board.containsKey(position);
```
null 비교가 아닌, 제공되는 메서드를 활용해보면 어떨까요?